### PR TITLE
Fix form validation error on One Off Contributions checkout page

### PR DESCRIPTION
**Pull request by jranks123**
## Why are you doing this?
There was a bug whereby the `isFormEmpty` prop for the `oneOffContributionsPayment` component (and the inline version of the page) was being incorrectly set, if the user isn't signed in. This is because the check to see if the name field is empty was `state.page.user.fullName === ''`. But if the user isn't signed in, `state.page.user.fullName` doesn't get set until the user types something into the name field, so `state.page.user.fullName` returns `undefined`. 

This fix changes the check to `!state.page.user.fullName`, and names the varaible `isFormEmpty` to `areAnyRequiredFieldsEmpty`, which better describes the field.

### CODE REVIEW

File: assets/pages/oneoff-contributions/components/oneoffContributionsPayment.jsx [hunk 1]
```diff
@@ -125,7 +125,7 @@ function OneoffContributionsPayment(props: PropTypes, context) {
           context.store.getState,
         )}
         canOpen={formValidation(
```
Comment (joelochlann) [inline assets/pages/oneoff-contributions/components/oneoffContributionsPayment.jsx]: Should you not change the name of the `isFormEmpty` param in `formValidation`? It no longer means what it says

Comment (joelochlann) [reply assets/pages/oneoff-contributions/components/oneoffContributionsPayment.jsx]: line 88 in `oneoffInlineContributionsPayment.jsx`

**Comment by joelochlann** [comment]
Not in your changes @jranks123, but I have to say I'm having trouble making sense of that `formValidation` function. Why is a function called `error` called when the form is valid?? Or am I completely misreading it??

```javascript

// If the form is valid, calls the given callback, otherwise sets an error.
function formValidation(
  isFormEmpty: boolean,
  validEmail: boolean,
  error: ?string => void,
): Function {

  return (): boolean => {

    if (!isFormEmpty && validEmail) {
      if (error) {
        error(null);
      }
      return true;
    }

    if (error) {
      if (isFormEmpty) {
        error('Please fill in all the fields above.');
      } else {
        error('Please fill in a valid email address.');
      }
    }
    return false;
  };

}

```

File: assets/pages/oneoff-contributions/components/oneoffContributionsPayment.jsx (hunk 1 above)
Comment (jranks123) [reply assets/pages/oneoff-contributions/components/oneoffContributionsPayment.jsx]: 👍

File: assets/pages/oneoff-contributions/components/oneoffInlineContributionsPayment.jsx [hunk 2]
```diff
@@ -85,22 +85,22 @@ function mapDispatchToProps(dispatch: Dispatch<OneOffCheckoutAction | StripeInli
 
 // If the form is valid, calls the given callback, otherwise sets an error.
 function formValidation(
-  isFormEmpty: boolean,
+  areAnyRequiredFieldsEmpty: boolean,
```
Comment (joelochlann) [inline assets/pages/oneoff-contributions/components/oneoffInlineContributionsPayment.jsx]: Out of scope for this PR, but I feel a clearer and more general abstraction for form validation could be to pass in the required fields and have them checked for truthiness inside the validation function.

Alternatively, an ideal solution would be to use the HTML5 constraint validation API!

https://developer.mozilla.org/en-US/docs/Learn/HTML/Forms/Form_validation#Validating_forms_using_JavaScript

https://caniuse.com/#search=constraint%20validation%20API

**Comment by Ap0c** [comment]
@joelochlann that function should probably be called `setError`, I believe in the happy case it's setting the error to `null`.

On your other point, I think it would be a great idea to standardise on form validation.

**Comment by jranks123** [comment]
@Ap0c nice, I've renamed `error` to `setError`

**Pull request merged**